### ISSUE TITLE
Delayed Discharges not matching onto admissions data

### ISSUE THREAD

**Issue by SwiftySalmon** [Public-Health-Scotland/source-linkage-files#882]
Only doing an `rbind` instead of `left_join` and then `rbind`. 

It'll be something like this however this is creating duplicates in the episode file due to several cases in DD matching on. Need to ask DD team about these and make a decision. 
   ` dplyr::left_join(linked_data %>%
                       dplyr::select(c(
                         chi, cij_marker, cij_start_date, cij_end_date, dd_responsible_lca,
                         delay_end_reason, primary_delay_reason,
                         secondary_delay_reason, cij_delay, dd_quality
                       )) %>%
                       dplyr::distinct() %>%
    dplyr::bind_rows(linked_data) `

**Comment by Jennit07** [comment]
Underlying data has two records for one admission e.g. different delay reason. Contact the DD team to check this

**Comment by lizihao-anu** [comment]
The original logic is first `inner_join` delay discharge data and episode files, and then process the data. Next, the processed combined data will simply bind back to episode files, since the recid is "DD" (delay discharge) and should not be `left_join`ed to data with any other recid, say AE2.

As for the quality of the data, there is an indicator varaible `dd_quality` to suggest the quality of this record matching to other records of other recid. 

As for the duplication, for year 1819, there are only 9 duplicated records (out of 28k DD records in total) and the duplication reason is for each record, there are two entries that the dd start or end date is in. For example, for AE2 data, for one patient, there could be two records, from 09/06/2018 to 20/06/2018 and from 20/06/2018 to 30/06/2018. The DD record for this patient is from 19/06/2018 to 20/06/2018. Then both AE2 records can be matched with the DD record. But the quality indicator `dd_quality` would suggest the first record is of better quality. 

In short, the logic is right. It might be better to contact DD team about keeping both records for duplication or keeping the one of best quality.
